### PR TITLE
Minor telecom code cleanup

### PR DESCRIPTION
**Pull request by silicons**
Removes STATION_Z define, we have that already as ZLEVEL_STATION.
Removes TELECOMM_Z, as we don't have a telecoms sat anymore
Removes anything that uses src.var instead of just the var's name.
Makes return 1/0 into return TRUE/FALSE

### CODE REVIEW

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 1]
```diff
@@ -219,14 +201,14 @@
...
-				temp = "<font color = #666633>-% [src] has been [src.toggled ? "activated" : "deactivated"].</font color>"
+				toggled = !toggled
+				temp = "<font color = #666633>-% [src] has been [toggled ? "activated" : "deactivated"].</font color>"
 				update_power()
 
 			/*
 			if("hide")
-				src.hide = !hide
-				temp = "<font color = #666633>-% Shadow Link has been [src.hide ? "activated" : "deactivated"].</font color>"
+				hide = !hide
```
Comment (lzimann) [inline code/game/machinery/telecomms/machine_interactions.dm]: Remove the commented code

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 2]
```diff
@@ -278,7 +260,7 @@
 			if(T)
 				temp = "<font color = #666633>-% Removed \ref[T] [T.name] from linked entities. %-</font color>"
 
-				// Remove link entries from both T and src.
+				// Remove link entries from both T and
```
Comment (lzimann) [inline code/game/machinery/telecomms/machine_interactions.dm]: Fix the comment!

File: code/game/machinery/telecomms/telecomunications.dm [hunk 3]
```diff
@@ -124,13 +124,13 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 	..()
 
 /obj/machinery/telecomms/proc/is_freq_listening(datum/signal/signal)
-	// return 1 if found, 0 if not found
+	// return TRUE if found, 0 if not found
```
Comment (lzimann) [inline code/game/machinery/telecomms/telecomunications.dm]: `FALSE` if not found

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 4]
```diff
@@ -295,8 +272,8 @@
 				if(!(src in T.links))
 					T.links.Add(src)
 
-				if(!(T in src.links))
-					src.links.Add(T)
+				if(!(T in links))
+					links.Add(T)
```
Comment (Cyberboss) [inline code/game/machinery/telecomms/machine_interactions.dm]: +=

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**